backend/src/main/java/de/bund/digitalservice/ris/norms/domain/entity/CharacterRange.java

```diff
@@ -54,6 +59,121 @@ public boolean isValidCharacterRange() {
     return pattern.matcher(characterRange()).matches() && isEndGreaterStart();
   }
 
+  /**
+   * Finds the text-content of the range within the given node.
+   *
+   * @param node the node in which the range should be found
+   * @return the text-content of the range
+   */
+  public String findTextInNode(Node node) {
+    var text =
+        getNodesInRange(node).stream().map(Node::getTextContent).collect(Collectors.joining());
+    node.normalize();
+    return text;
+  }
+
+  /**
+   * Splits the given text node into multiple text nodes and returns the text node that includes the
+   * range from start to end of the text content.
+   *
+   * <p>A call to Node::normalize undoes this.
+   *
+   * @param textNode A text node
+   * @param start start of the text range within the node
+   * @param end end (excluding) of the text range within the node
+   * @return a text node that represents exactly the range
+   */
+  private Node extractTextNodeForRange(Node textNode, int start, int end) {
```

Comment:
Why manipulate the passed textNode if the goal of the method is to extract and return only the text node for the given range? The passed node will be manipulated and splitted into several text nodes and this will modify the XML right?

Reply:
Yes this does modify the original XML, but that is actually the idea. With this we can then take the text node returned from this method and eg. replace it with a new node. Which is just the thing we need for applying change commands. Also the side effect here is quite minor as it can be reversed by normalizing (Node::normalize) the xml.